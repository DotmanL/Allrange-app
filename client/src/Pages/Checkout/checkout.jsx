import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import StripeCheckoutButton from '../../components/stripe-button/stripe-button';
import CheckoutItem from '../../components/checkout-item/checkout-item';
import Paystack from '../../components/paystack/paystack-button'

import {
  CheckoutPageContainer,
  CheckoutHeaderContainer,
  HeaderBlockContainer,
  TotalContainer,
  WarningContainer, 
  EmptyCartContainer
} from './checkout.styles';


import {
  selectCartItems,
  selectCartTotal
} from '../../redux/cart/cart.selectors';

 

export const CheckoutPage = ({ cartItems, total, }) => (
  <CheckoutPageContainer>
    <CheckoutHeaderContainer>
      <HeaderBlockContainer>
        <span>Product</span>
      </HeaderBlockContainer>
      <HeaderBlockContainer>
        <span>Description</span>
      </HeaderBlockContainer>
      <HeaderBlockContainer>
        <span>Quantity</span>
      </HeaderBlockContainer>
      <HeaderBlockContainer>
        <span>Price</span>
      </HeaderBlockContainer>
      <HeaderBlockContainer>
        <span>Remove</span>
      </HeaderBlockContainer>
    </CheckoutHeaderContainer>

    {cartItems.map(cartItem => (
      <CheckoutItem key={cartItem.id} cartItem={cartItem} />
    ))}
    <TotalContainer>TOTAL:  â‚¦{total}</TotalContainer>
    
    {cartItems.length ? (
  <Paystack price={total} />
  ) : ( 
<EmptyCartContainer>Fill your Cart before you can make payments</EmptyCartContainer>
  )}
  


    {cartItems.length ? (
    <StripeCheckoutButton price={total} />
    ) : ( 
      null
    )}

<WarningContainer>
      <strong>*Please use the following test credit card for payments*</strong>
      <br />
      <strong> 5555 5555 5555 4444 - Exp: 01/22 - CVV:123</strong>
    </WarningContainer>
  </CheckoutPageContainer>





);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal,
});

export default connect(mapStateToProps)(CheckoutPage);



 