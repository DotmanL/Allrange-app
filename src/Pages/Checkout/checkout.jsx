import React from 'react';
import {connect} from 'react-redux';
import {createStructuredSelector} from 'reselect'

import {selectCartItems,  selectCartTotal} from '../../redux/cart/cart.selectors'
import CheckOutItem from '../../components/checkout-item/checkout-item';
import StripeCheckoutButton from '../../components/stripe-button/stripe-button';
import './checkout.scss';

const CheckOutPage = ( {cartItems, total}) => (
 <div className='checkout-page'>
      <div className='checkout-header'>
      <div className='header-block'>
      <span>Product</span>
      </div>
      <div className='header-block'>
      <span>Description</span>
      </div>
      <div className='header-block'>
      <span>Quantity</span>
      </div>
      <div className='header-block'>
      <span>Price</span>
      </div>
      <div className='header-block'>
      <span>Remove</span>
      </div>
   </div>
   {
       cartItems.map(cartItem =>(
        <CheckOutItem key ={cartItem.id} cartItem={cartItem}/>
        ))}
   <div className='total'>Total: â‚¦{total}</div>
<div className='test-warning'>
   <strong>*Please use the following test credit card for payments*</strong> 
<br />
<strong>5555 5555 5555 4444 - Exp: 01/22 - CVV:123</strong>
</div>
<StripeCheckoutButton price={total} />
   </div>
)




const mapStateToProps = createStructuredSelector ({
    cartItems: selectCartItems,
    total: selectCartTotal
})


export default connect( mapStateToProps)
    (CheckOutPage);