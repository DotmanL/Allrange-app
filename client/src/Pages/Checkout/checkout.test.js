import React from 'react';
import { shallow } from 'enzyme';
import  { CheckoutPage }  from './checkout'
import toJson from "enzyme-to-json";

let wrapper;
beforeEach(() => {
  const mockProps = {
    cartItems: [],
    total: 100
  };

  wrapper = shallow(<CheckoutPage {...mockProps} />);
});

it('should render CheckoutPage component', () => {
  expect(
  toJson(wrapper)
).toMatchSnapshot();
})