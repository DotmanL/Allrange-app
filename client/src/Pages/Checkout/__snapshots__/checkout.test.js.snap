// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CheckoutPage component 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <span>
        Product
      </span>
    </styled.div>
    <styled.div>
      <span>
        Description
      </span>
    </styled.div>
    <styled.div>
      <span>
        Quantity
      </span>
    </styled.div>
    <styled.div>
      <span>
        Price
      </span>
    </styled.div>
    <styled.div>
      <span>
        Remove
      </span>
    </styled.div>
  </styled.div>
  <styled.div>
    TOTAL:  â‚¦
    100
  </styled.div>
  <styled.span>
    Fill your Cart before you can make payments
  </styled.span>
  <styled.div>
    <strong>
      *Please use the following test credit card for payments*
    </strong>
    <br />
    <strong>
       5555 5555 5555 4444 - Exp: 01/22 - CVV:123
    </strong>
  </styled.div>
</styled.div>
`;
